eo-parser/src/main/java/org/eolang/parser/ParsingErrors.java

```diff
@@ -81,18 +83,40 @@ public void syntaxError(
         final String msg,
         final RecognitionException error
     ) {
-        this.errors.add(
-            new ParsingException(
-                String.format(
-                    "[%d:%d] %s: \"%s\"",
-                    line, position, msg,
-                    // @checkstyle AvoidInlineConditionalsCheck (1 line)
-                    this.lines.size() < line ? "EOF" : this.lines.get(line - 1)
-                ),
-                error,
-                line
-            )
-        );
+        // @checkstyle MethodBodyCommentsCheck (20 lines)
+        // @todo #3332:30min Add more specific error messages.
+        //  Currently we write just "error: no viable alternative at input" for all errors.
+        //  It's better to use {@link #recognizer} to get more specific error messages.
```

Comment:
@volodya-lombrozo who is `#recognizer`? If I would be the resolver of this puzzle, I would be very puzzled.